File: BE/PE_SBA301_SP25_BE_StudentCode/src/main/java/com/de180293/pe_sba301_sp25_be_de180293/service/AuthService.java
package com.de180293.pe_sba301_sp25_be_de180293.service;

import com.de180293.pe_sba301_sp25_be_de180293.dto.request.LoginRequest;
import com.de180293.pe_sba301_sp25_be_de180293.dto.request.RefreshTokenRequest;
import com.de180293.pe_sba301_sp25_be_de180293.dto.request.RegisterRequest;
import com.de180293.pe_sba301_sp25_be_de180293.dto.response.AuthResponse;
import com.de180293.pe_sba301_sp25_be_de180293.dto.response.CustomerResponse;

public interface AuthService {
    CustomerResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    CustomerResponse me(String email);
    AuthResponse refresh(RefreshTokenRequest request);
}

File: LAC_VN/FE/src/api/authService.ts
import axiosClient from './axiosClient';
import { 
  AuthResponse, 
  OtpResponse, 
  LoginRequest, 
  RegisterRequest, 
  OTPRequest, 
  VerifyOTPRequest 
} from '../types/auth.types';
import { ApiResponse } from '../types/product.types';

const authService = {
  sendOtp: async (email: string): Promise<OtpResponse> => {
    const request: OTPRequest = { email };
    const response = await axiosClient.post<ApiResponse<OtpResponse>>('/auth/send-otp', request);
    return response.data.data;
  },

  verifyOtp: async (email: string, otp: string): Promise<OtpResponse> => {
    const request: VerifyOTPRequest = { email, otp };
    const response = await axiosClient.post<ApiResponse<OtpResponse>>('/auth/verify-otp', request);
    return response.data.data;
  },

  login: async (email: string, password: string): Promise<AuthResponse> => {
    const request: LoginRequest = { email, password };
    const response = await axiosClient.post<ApiResponse<AuthResponse>>('/auth/login', request);
    return response.data.data;
  },

  register: async (data: RegisterRequest): Promise<AuthResponse> => {
    const response = await axiosClient.post<ApiResponse<AuthResponse>>('/auth/register', data);
    return response.data.data;
  },

  logout: async (): Promise<void> => {
    localStorage.removeItem('accessToken');
    localStorage.removeItem('user');
  },
};

export default authService;
